### PR TITLE
Rename .rn to .native

### PR DESCRIPTION
**Pull request by trivikr**
*Issue #, if available:*
Refs: https://github.com/aws/aws-sdk-js-v3/issues/990

*Description of changes:*
Rename .rn to .native

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptDependency.java [hunk 1]
```diff
@@ -47,7 +47,7 @@
 
     AWS_SDK_STREAM_COLLECTOR_NODE("dependencies", "@aws-sdk/stream-collector-node", "^1.0.0-alpha.1", true),
     AWS_SDK_STREAM_COLLECTOR_BROWSER("dependencies", "@aws-sdk/stream-collector-browser", "^1.0.0-alpha.1", true),
-    AWS_SDK_STREAM_COLLECTOR_RN("dependencies", "@aws-sdk/stream-collector-rn", "^1.0.0-alpha.0", true),
+    AWS_SDK_STREAM_COLLECTOR_RN("dependencies", "@aws-sdk/stream-collector-native", "^1.0.0-alpha.0", true),
```
Comment (AllanFly120) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptDependency.java]: Why is this necessary?

Comment (trivikr) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptDependency.java]: Since we're preferring `.native` vs `.rn` extensions we should also move from `-rn` to `-native` package names.

Similar to `.browser` extension and `-browser` package name

Comment (trivikr) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptDependency.java]: Relevant commit for updating package names in codegen https://github.com/aws/aws-sdk-js-v3/pull/1025/commits/bb1e16bf63ba2a4446d4f962952b0c8e71611c14

Comment (AllanFly120) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptDependency.java]: I'm not apposing to it, but I don't think it's necessary either.🙂
